ruby/generate-short-url/index.rb

```diff
@@ -0,0 +1,71 @@
+require 'json'
+require 'uri'
+require 'net/http'
+
+VALID_PROVIDERS = %w[bitly tinyurl].freeze
+API_URLS = {
+  'bitly' => 'https://api-ssl.bitly.com/v4/shorten',
+  'tinyurl' => 'https://api.tinyurl.com/create'
+}.freeze
+
+def main(req, res)
+  payload = req.payload
```

Comment:
To make this compatible with Appwrite functions, would you please test this with payload as a JSON string and parse it here. For example:

```shell
curl -H "X-Internal-Challenge: secret-key" \
    -H "Content-Type: application/json" \
    -X POST http://localhost:3000/ \
    -d '{"payload": "{\"url\": \"https://google.com\"}"}' 
```